### PR TITLE
feat(styles): create validation component

### PR DESCRIPTION
**Pull request by myrta2302**

### CODE REVIEW

File: packages/styles/src/components/validation.scss [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   validation msg has class -> validation-msg
+*/
+
+.validation-msg {
+  display: none;
+}
+
+.validated {
+  input[type='radio'].invalid ~ .validation-msg,
+  input[type='checkbox'].invalid ~ .validation-msg {
```
Comment (alizedebray) [inline packages/styles/src/components/validation.scss:28]: These inputs should also have an icon when they are valid, this is so that color-blind people can get the same information as others.

File: packages/styles/src/components/validation.scss [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    &:before {
+      height: $post-radio-button-icon-height-inner;
+      width: $post-radio-button-icon-width-inner;
+      display: inline-block;
+      margin-right: $post-validation-list-gap-inline;
+      // @include icons.icon(2413);
+      background-image: url('error-icon.svg');
+      background-size: contain;
```
Comment (alizedebray) [inline packages/styles/src/components/validation.scss:36]: We usually use `mask-image` for the icons so that they inherit the text color: 
```suggestion
      mask-image: url('error-icon.svg');
      backround-color: currentColor;
```
You can have a look ate [this mixin](https://github.com/swisspost/design-system/blob/main/packages/styles/src/mixins/_icons.scss#L10). It will be used as soon as the new icons are properly defined.

File: packages/styles/src/components/validation.scss [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.validated {
+  input[type='radio'].invalid ~ .validation-msg,
+  input[type='checkbox'].invalid ~ .validation-msg {
+    &:before {
+      height: $post-radio-button-icon-height-inner;
+      width: $post-radio-button-icon-width-inner;
+      display: inline-block;
+      margin-right: $post-validation-list-gap-inline;
```
Comment (alizedebray) [inline packages/styles/src/components/validation.scss:33]: Please use `block`and `inline` properties instead of `top`, `bottom`, `right` and `left`. https://developer.mozilla.org/en-US/docs/Web/CSS/margin-inline-end

File: packages/styles/src/components/validation.scss [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      // @include icons.icon(2413);
+      background-image: url('error-icon.svg');
+      background-size: contain;
+      content: '';
+      flex-shrink: 0;
+    }
+  }
+  .validation-msg,
+  .validation-msg {
+    display: inline-flex;
```
Comment (alizedebray) [inline packages/styles/src/components/validation.scss:43]: This class is meant to be used on a paragraph which is a block element.
```suggestion
    display: flex;
```

File: packages/styles/src/components/validation.scss [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    align-items: center;
+    width: 100%;
+    margin: 0;
+    padding: var(--post-input-spacing-padding-block-text-assist, 6px)
+      var(--post-input-spacing-padding-inline-text-assist, 2px);
+    font-family: $post-device-font-family-default;
+    font-size: $post-device-font-size-9;
+    font-weight: $post-device-font-weight-regular;
```
Comment (alizedebray) [inline packages/styles/src/components/validation.scss:51]: These are the default font-family and font-weight so they are inherited from the body:
```suggestion
    font-size: $post-device-font-size-9;
```

File: packages/styles/src/components/validation.scss [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+@use './../mixins/icons';
+
+$post-validation-error: #b20000;
+$post-validation-success: #107800;
+$post-device-font-family-default: 'Swiss Post Sans v005';
+$post-device-font-size-9: 14px;
+$post-device-font-weight-regular: 400;
+$post-radio-button-icon-width-inner: 24px;
+$post-radio-button-icon-height-inner: 24px;
+$post-validation-list-gap-inline: 8px;
```
Comment (alizedebray) [inline packages/styles/src/components/validation.scss:13]: Can you use the `token.get` function to set these variables? https://github.com/swisspost/design-system/blob/main/packages/styles/src/functions/_tokens.scss#L31

It will help implementing the real tokens in the end. 
```suggestion
@use './../functions/tokens';

tokens.$default-map: (
  post-validation-error: #b20000,
  post-validation-success: #107800,
  post-device-font-family-default: 'Swiss Post Sans v005',
  post-device-font-size-9: 14px,
  post-device-font-weight-regular: 400,
  post-radio-button-icon-width-inner: 24px,
  post-radio-button-icon-height-inner: 24px,
  post-validation-list-gap-inline: 8px,
);
```

File: packages/styles/index.html [hunk 7]
```diff
@@ -31,6 +31,65 @@ <h1>Styles package playground</h1>
...
+        <select class="form-select" aria-label="select example" required>
+          <option value="">Open this select menu</option>
+          <option value="1">One</option>
+          <option value="2">Two</option>
+          <option value="3">Three</option>
+        </select>
+        <p class="valid-feedback">You’ve selected an option.</p>
+        <p class="invalid-feedback">Please select an option.</p>
+      </div>
+      <fieldset class="form-check" required>
```
Comment (alizedebray) [inline packages/styles/index.html:67]: Did you find this fieldset structure in the docs? If so, what is the link?

Comment (myrta2302) [reply packages/styles/index.html:67]: i think i took it from the documentation from the inline example :https://design-system.post.ch/?path=/docs/151242aa-a074-4a55-a81c-db597c83cdad--docs

**Comment by myrta2302** [comment]
- Icon needs negative margin top for centering
- SVG in unofficial map dont work as expected for smaller screens /  also error svg looks 'cut' on the right

**Comment by myrta2302** [comment]
- Margin negative to align icon with text /add margin to p also
- Dark mode cardcontrol possible accessibility issue
- Icons from unofficial map not responding well to breakpoints
- Icon danger is 'cut' on the right

File: packages/styles/src/components/validation.scss [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+.was-validated fieldset:has(:invalid),
+fieldset:has(.is-invalid) {
+  ~ .invalid-feedback {
+    color: tokens.get('post-validation-error');
+
+    &::before {
+      @include icon-mixin.icon('val-error');
+    }
+  }
+}
```
Comment (alizedebray) [inline packages/styles/src/components/validation.scss:127]: Can you refactor the code to avoid duplication?

File: packages/documentation/src/stories/components/forms/validation/validation.stories.ts [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+  tags: ['package:HTML'],
+  parameters: {
+    badges: [],
+    design: {
+      type: 'figma',
+      url: 'https://www.figma.com/design/JIT5AdGYqv6bDRpfBPV8XR/Foundations-%26-Components-Next-Level?node-id=577-14513&t=uKEtTo9BEaPpKSV5-1',
+    },
+  },
+  args: {
+    validation: 'null',
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.stories.ts:17]: For most of the stories the validation "Default" is not selected.

File: packages/documentation/src/stories/components/forms/validation/validation.stories.ts
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.stories.ts:1]: It would be better to use floating labels for the text input, select and textarea since that would become the default behavior.

**Review by alizedebray — CHANGES_REQUESTED** [review]

File: packages/styles/src/components/validation.scss [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+/*
+ Checkbox and Radio box specific classes
+*/
+input[type='checkbox'],
+input[type='radio'],
+input[type='range'] {
+  ~ .invalid-feedback,
+  ~ .valid-feedback {
+    margin-top: 1px;
```
[the remark this answers is not in the document]
Comment (myrta2302) [reply packages/styles/src/components/validation.scss:51]: This is the margin that "corrects" the icon's negative margin.

File: packages/styles/src/components/validation.scss (hunk 8 above)
Comment (myrta2302) [reply packages/styles/src/components/validation.scss]: removed fieldset related styles

**Comment by myrta2302** [comment]
I did not remove validation from the individual form components, because each component implementation was different. 
I added the data-color-scheme attribute instead.

**Comment by myrta2302** [comment]
Removed Card Control validation example and teste since its going to be deprecated.

File: packages/styles/src/variables/_icons.scss [hunk 11]
```diff
@@ -7,7 +7,7 @@
 
 $icon-size-default: 1.5rem !default;
 
-$svg-pre-path: "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='32' height='32' fill='none' viewBox='0 0 32 32'%3E%3Cpath d='";
+$svg-pre-path: "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg'  viewBox='0 0 32 32'%3E%3Cpath d='";
```
Comment (alizedebray) [inline packages/styles/src/variables/_icons.scss:10]: Is this change necessary?

File: packages/documentation/src/stories/components/forms/validation/validation.stories.ts [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+};
+
+export default meta;
+
+type Story = StoryObj;
+export const Default: Story = {};
+
+export const Checkbox: Story = {
+  args: {
+    validation2: 'is-invalid',
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.stories.ts:42]: Please use meaningful names:
```suggestion
    checkboxValidation: 'is-invalid',
```

File: packages/documentation/src/stories/components/forms/validation/validation.stories.ts [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+  description: 'Controls the validation state appearance of the component.',
+  control: {
+    type: 'radio',
+    labels: {
+      'null': 'Default',
+      'is-valid': 'Valid',
+      'is-invalid': 'Invalid',
+    },
+  },
+  options: ['null', 'is-valid', 'is-invalid'],
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.stories.ts:29]:
```suggestion
  options: ['is-invalid', 'is-valid', 'null'],
```

File: packages/documentation/src/stories/components/forms/validation/validation.stories.ts [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+  },
+  render(args: Args) {
+    const isValidationSet = args.validation2 !== 'null';
+    const isValid = args.validation2 === 'is-valid';
+    let ariaInvalid;
+    if (isValidationSet) {
+      ariaInvalid = !isValid;
+    } else {
+      ariaInvalid = nothing;
+    }
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.stories.ts:55]: Why not using a ternary operator here: 
```suggestion
    const ariaInvalid = isValidationSet ? !isValid : nothing;
```

File: packages/documentation/src/stories/components/forms/validation/validation.stories.ts
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.stories.ts:1]: Can you refactor this file to avoid code duplication? 
You could for example have a function the returns the error messages template, and another that takes the args as an input and return extrapolated arguments used by all the stories (`isValidationSet`, `isValid`, `ariaInvalid`, ...)

File: packages/documentation/src/stories/components/forms/validation/validation.snapshot.stories.ts
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.snapshot.stories.ts:1]: Can you use the renderer from the stories to make this file more maintainable?

**Review by alizedebray — CHANGES_REQUESTED** [review]

File: packages/documentation/src/stories/components/forms/input/input.docs.mdx
Comment (leagrdv) [inline packages/documentation/src/stories/components/forms/input/input.docs.mdx:1]: The styles import are not correct anymore since it's not `form-feedback.scss` but `validation.scss` now. (same goes for the other form elements docs.mdx)

File: packages/documentation/src/stories/components/forms/textarea/textarea.snapshot.stories.ts [hunk 15]
```diff
@@ -47,27 +47,9 @@ export const Textarea: Story = {
...
-                              ${Object.entries(context.argTypes.size.control.labels)
-                                .filter(([key]) => key === args.size)
-                                .map(s => s[1])}
-                            </h4>
-                          `
-                        : ''}
-                      <div>${meta.render?.({ ...context.args, ...args }, context)}</div>
-                    </div>
-                  `;
+                  return html` <div>${meta.render?.({ ...context.args, ...args }, context)}</div> `;
```
Comment (leagrdv) [inline packages/documentation/src/stories/components/forms/textarea/textarea.snapshot.stories.ts:50]: I think you removed too much here. All of the floating labels variants have disappeared.

**Review by leagrdv — CHANGES_REQUESTED** [review]

File: packages/documentation/src/stories/components/forms/validation/validation.stories.ts [hunk 16]
```diff
@@ -0,0 +1,312 @@
...
+      : nothing}
+    ${invalidFeedbackId
+      ? html`<p id="${invalidFeedbackId}" class="invalid-feedback">Invalid message.</p>`
+      : nothing}
+  `;
+}
+
+const meta: MetaComponent = {
+  id: '1aa900d9-aa65-4ae0-b8cd-e6cca6cc3472',
+  title: 'Components/Forms/Validation',
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/forms/validation/validation.stories.ts:58]:
```suggestion
  title: 'Components/Form Validation',
```

File: packages/styles/src/components/form-validation.scss [hunk 17]
```diff
@@ -1,116 +0,0 @@
...
-@use './../variables/spacing';
-@use './../variables/type';
-@use './../mixins/icons';
-
-.invalid-feedback,
-.valid-feedback {
-  display: none;
-  align-items: center;
-  width: 100%;
-  margin-block: form-validation.$form-feedback-margin-top 0;
```
Comment (leagrdv) [inline packages/styles/src/components/form-validation.scss:23]: You've deleted a lot of variables from the _form-validation.scss variables file, make sure to delete the ones that are no longer used anywhere. You can probably even delete the file if none are used anymore

**Review by leagrdv — CHANGES_REQUESTED** [review]

**Review by leagrdv — APPROVED** [review]
